src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/HistoryTabPanel.kt

```diff
@@ -1,41 +1,59 @@
 package com.atlassian.performance.tools.jiraactions.api.page
 
 import org.openqa.selenium.By
+import org.openqa.selenium.Keys
 import org.openqa.selenium.WebDriver
+import org.openqa.selenium.interactions.Actions
 import org.openqa.selenium.support.ui.ExpectedConditions
 import java.time.Duration
 
 class HistoryTabPanel(
     private val driver: WebDriver
 ) {
     private val loadMoreEntriesLocator = By.cssSelector("button.show-more-changehistory-tabpanel")
+    private val duration: Duration = Duration.ofMinutes(2)
 
-    fun loadAllHistoryEntries(): HistoryTabPanel {
+
+    fun showAllHistoryEntries(): HistoryTabPanel {
         if (driver.isElementPresent(loadMoreEntriesLocator)) {
-            val loadMoreButton = driver.findElement(loadMoreEntriesLocator)
-            loadMoreButton.click()
-            driver.wait(
-                timeout = Duration.ofSeconds(60),
-                condition = ExpectedConditions.stalenessOf(loadMoreButton)
-            )
-        } else {
-            driver.wait(
-                timeout = Duration.ofSeconds(60),
-                condition = ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(".issuePanelWrapper.loading"))
-            )
+            return showMoreEntriesIfNeeded()
         }
+        driver.wait(
+            duration,
+            ExpectedConditions.invisibilityOfElementLocated(By.cssSelector(".issuePanelWrapper.loading"))
+        )
         return this
     }
 
     fun waitForActive(): HistoryTabPanel {
         driver.wait(
-            Duration.ofSeconds(45),
+            duration,
             ExpectedConditions.elementToBeClickable(By.id("changehistory-tabpanel"))
         ).click()
         driver.wait(
-            Duration.ofSeconds(60),
+            duration,
             ExpectedConditions.presenceOfElementLocated(By.cssSelector("#changehistory-tabpanel.active-tab"))
         )
         return this
     }
+
+    private fun showMoreEntriesIfNeeded(): HistoryTabPanel {
+        do {
+            val loadMoreButton = driver.findElement(loadMoreEntriesLocator)
+
+            Actions(driver)
+                .keyDown(Keys.SHIFT)
+                .click(loadMoreButton)
+                .keyUp(Keys.SHIFT)
+                .build()
+                .perform()
+            driver.wait(
+                duration,
+                ExpectedConditions.stalenessOf(loadMoreButton)
+            )
+        } while (
+            driver.isElementPresent(loadMoreEntriesLocator)
```

Comment:
We should limit users patience here, so that we won't fall into infinite loop here (e.g. button not working due to network issue). I think that after 5-10 clicks I would give up.

Reply:
With adding of this patience I wonder if we shouldn't fail the action in case of not loading all of the comments.

Alternatively at least maybe we should add number of comments loaded to the `observation` parameter of `meter.measure()`. This would be to make it possible to ensure that we are comparing apples to apples, e.g. if one version if Jira, doesn't cheat by loading less data.

Reply:
Same for `CommentTabPanel#showAllCommentsInJira9`

Reply:
I like the idea of adding the number of loaded elements to the observation, but would it be okay for you to implement it later in a separate PR?

Reply:
@ewefie I'm specifically using "Consider..." in the action checkbox text whenever I'm ok with not doing this.

Feel free to skip this feedback if you think it's not valid for this PR.

If you think it could still be done later then you could create a [JPERF](https://ecosystem.atlassian.net/browse/JPERF) issue and address it later even by yourself.